Utils/Services/ImagePickerService.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ImagePickerService.swift
+//  Utils
+//
+//  Created by Nahuel Gladstein on 6/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import ReactiveSwift
+import UIKit
+import AVFoundation
+import Photos
+import Result
+
+public enum ImagePickerServiceError: Error {
+    
+    case sourceTypeNotAvailable
+    
+}
+
+public protocol ImagePickerServiceType {
```

Comment:
About the type, I guess we could use the internal type. I'm a little reticent because they may not be friendly in name, or have the format we use in the rest of our code, and making it our gives us more flexibility to decide over it afterwards.